lib/OrdinaryDiffEqDifferentiation/src/derivative_utils.jl

```diff
@@ -39,7 +39,39 @@ function calc_tderivative!(integrator, cache, dtd1, repeat_step)
             else
                 tf.uprev = uprev
                 tf.p = p
-                derivative!(dT, tf, t, du2, integrator, cache.grad_config)
+                alg = unwrap_alg(integrator, true)
+                #derivative!(dT, tf, t, du2, integrator, cache.grad_config)
+                autodiff_alg = alg_autodiff(alg)
+
+                autodiff_alg = if autodiff_alg isa AutoSparse
+                    ADTypes.dense_ad(autodiff_alg)
+                else
+                    autodiff_alg
+                end
```

Comment:
```suggestion
                autodiff_alg = ADTypes.dense_ad(autodiff_alg)
```
As soon as ADTypes v1.14 is out, we can do this everywhere